dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java

```diff
@@ -42,14 +42,15 @@ public static Channel create(
         // n.b. This becomes cleaner once we support reloadable channels, the queue can be created first, and
         // each limited channel can be created later and passed a method reference to the queued channel.
         DeferredLimitedChannelListener queueListener = new DeferredLimitedChannelListener();
-        List<LimitedChannel> limitedChannels = channels.stream()
+        List<CompositeLimitedChannel> limitedChannels = channels.stream()
                 // Instrument inner-most channel with metrics so that we measure only the over-the-wire-time
                 .map(channel -> new InstrumentedChannel(channel, clientMetrics))
                 // TracedChannel must wrap TracedRequestChannel to ensure requests have tracing headers.
                 .map(TracedRequestChannel::new)
                 .map(channel -> new TracedChannel(channel, "Dialogue-http-request"))
-                .map(LimitedChannelAdapter::new)
+                .map(CompositeLimitedChannelAdapter::new)
                 .map(concurrencyLimiter(config, clientMetrics))
+                .map(channel -> new BlacklistingChannel(channel, config.failedUrlCooldown(), queueListener))
```

Comment:
after any failure, this will put the channel into probation mode by default (failedUrlCooldown == 0 using the default configuration).
This could be problematic for timelock, perhaps we should check if clientQos is disabled for this?

Reply:
ya good catch